src/Asm89/Stack/CorsService.php

```diff
@@ -98,16 +98,16 @@ public function addActualRequestHeaders(Response $response, Request $request)
 
     public function handlePreflightRequest(Request $request)
     {
-        if (true !== $check = $this->checkPreflightRequestConditions($request)) {
-            return $check;
-        }
+        $response = new Response();
 
-        return $this->buildPreflightCheckResponse($request);
+        return $this->addPreflightRequestHeaders($response, $request);
     }
 
-    private function buildPreflightCheckResponse(Request $request)
+    public function addPreflightRequestHeaders(Response $response, Request $request)
     {
-        $response = new Response();
+        if (true !== $check = $this->checkPreflightRequestConditions($request)) {
```

Comment:
Conflicted about moving this check here. The method is called `addPreflightRequestHeaders()`, but is now checking the request for the right headers. If the request is not valid for a pass we still end up returning a different response object.